### PR TITLE
Unlock pending block queue if insertion errors

### PR DESCRIPTION
**Pull request by terencechain**
While working on 4844, I noticed when `insertBlockToPendingQueue` errors out, `pendingQueueLock` lock still holds. We should be releasing the lock

### CODE REVIEW

File: beacon-chain/sync/rpc_beacon_blocks_by_root.go [hunk 1]
```diff
@@ -25,6 +25,7 @@ func (s *Service) sendRecentBeaconBlocksRequest(ctx context.Context, blockRoots
 		}
 		s.pendingQueueLock.Lock()
 		if err := s.insertBlockToPendingQueue(blk.Block().Slot(), blk, blkRoot); err != nil {
+			s.pendingQueueLock.Unlock()
 			return err
 		}
 		s.pendingQueueLock.Unlock()
```
Comment (saolyn) [inline beacon-chain/sync/rpc_beacon_blocks_by_root.go:31]:
```suggestion
		s.pendingQueueLock.Lock()
		defer s.pendingQueueLock.Unlock()
		if err := s.insertBlockToPendingQueue(blk.Block().Slot(), blk, blkRoot); err != nil {
			return err
		}
```
This is a better pattern for you.

**Review by saolyn — CHANGES_REQUESTED** [review]
Forgot to request change

File: beacon-chain/sync/rpc_beacon_blocks_by_root.go (hunk 1 above)
Comment (terencechain) [reply beacon-chain/sync/rpc_beacon_blocks_by_root.go]: Thanks @saolyn, it's ready again!

**Review by saolyn — APPROVED** [review]
LGTM

**Pull request merged**